### ISSUE TITLE
FNN (Univariate)

### ISSUE THREAD

**Issue by FNUAreena** [FNUAreena/fall-2025-group9#83]
**Goal**

Train a **Feedforward Neural Network (FNN)** on TimeSeriesDataset `production_cost_total` time series and compare its fit to the LSTM Model.

** FNN Model**
- The FNN architecture is defined as `FeedForwardRegressor` in  
  [model.py](https://github.com/FNUAreena/fall-2025-group9/blob/main/src/component/univariate/model.py).

- It is created and trained in  
  [comparing_model.py](https://github.com/FNUAreena/fall-2025-group9/blob/main/src/component/univariate/comparing_model.py).

- Main settings:
  - Input dimension: `in_dim = WINDOW` (7 time steps).
  - Hidden size: `HIDDEN = 256`.
  - Dropout: `DROPOUT = 0.25`.

- Training configuration:
  - Device: CPU.
  - Optimizer: Adam with `LR_FNN = 0.001`.
  - Loss: Mean Squared Error (`nn.MSELoss()`).
  - Epochs: `EPOCHS_FNN = 100`.

- Training loop:
  - Convert `Xtr` and `ytr` to PyTorch tensors.
  - For 100 epochs:
    - Forward pass → compute loss → backpropagation → optimizer step.
    - Print training loss every 10 epochs for monitoring.

** Evaluation and FNN plot**
- After training:
  - Use the FNN to predict on `Xe`.
  - Inverse-transform predictions and true values back to original cost.

- Overall metrics for FNN :
  - **MSE (FNN)**: `39595.32`
  - **RMSE (FNN)**: `198.99`
  - **R² (FNN)**: `0.749`

  - X-axis: actual production cost.
  - Y-axis: FNN predicted production cost.
  - Dashed `y = x` line for perfect prediction.
  - Title: **“FNN Predicted vs Actual”**.
  - Saved to: `univariate/plots/fnn_model.png`  
    (file: [fnn_model.png](https://github.com/FNUAreena/fall-2025-group9/blob/main/src/component/univariate/plots/fnn_model.png)).

- The trained FNN weights are saved as:
  - `univariate/results/fnn_model.pth`

 **How to run**

```bash
cd src/component
python univariate/comparing_model.py
